### ISSUE TITLE
C++ warning: Control reaches end of non-void function

### ISSUE THREAD

**Issue by ubruhin** [slint-ui/slint#7864]
### Bug Description

The code below generates C++ code which causes the following compiler warning:

```
warning: control reaches end of non-void function [-Wreturn-type]
   24 |                         [&]() -> bool { if (true) { self->root_1_do_something.call(); } else { return self->root_1_do_something_else.call(); }}();
      |                                                                                                                                               ^
```

I'm not sure how harmful it is, but at least this time it's not a false-positive for sure as the generated code is clearly kind of invalid ;)

### Reproducible Code (if applicable)

```slint
export component AppWindow inherits Window {
    pure callback do-something();
    pure callback do-something-else() -> bool;

    TouchArea {
        clicked => {
            if true {
                do-something();
            } else {
                do-something-else();
            }
        }
    }
}
```

### Environment Details

- Slint Version: 1.10.0
- Platform/OS: GCC / Linux
- Programming Language: C++
- Backend/Renderer: Qt
